utils/oominfo/oomparser.go

```diff
@@ -0,0 +1,200 @@
+package main
+
+import (
+  "bufio"
+  "fmt"
+  "log"
+  "os"
+  "strings"
+  "os/exec"
+  "time"
+  "regexp"
+  "io"
+  "errors"
+)
+
+// struct to hold a list of oomInstance objects
+type oomLogList struct {
+	oomInstances []*oomInstance
+}
+
+// struct that contains information related to an OOM kill instance
+type oomInstance struct {
+	pid string
+	processName string
+	timeOfDeath time.Time
+	containerName containerLevel
```

Comment:
We use absolute path container names (so it can be a string):

/top-level-container
/top/with-a-subcontainer
/top/child/even-deeper-subcontainer